src/Analyzers/MSTest.GlobalConfigsGenerator/AnalyzerSeverityDecider.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.CodeAnalysis;
+
+// NOTE for our current matrix.
+// We decide whether to include a rule in a specific mode or not based on two factors.
+// 1. IsEnabledByDefault (true/false)
+// 2. DefaultSeverity (Info, Warn) - note: we currently don't use severity Hidden or Error.
+// So, we have 4 possible combinations.
+// 1. IsEnabledByDefault = true, DefaultSeverity = Warn (we include that in All, Recommended, and Default)
+// 2. IsEnabledByDefault = true, DefaultSeverity = Info (we include that in All and Recommended)
+// 3. IsEnabledByDefault = false, DefaultSeverity = Warn (we don't combine Warn with not enabled by default)
+// 4. IsEnabledByDefault = false, DefaultSeverity = Info (we include that in All only)
```

Comment:
Not a bad strategy, I am wondering if we should have some extra tags on each rules to help with that. WDYT?

Reply:
@Evangelink You mean have define the minimum mode in each rule? Tags like `AllOnly`, `RecommendedAndAll`, `DefaultAndRecommendedAndAll` (putting naming aside)? Then each rule has to explicitly declare the right tag?

Or did you mean something like "decide the applicable modes automatically based on `IsEnabledByDefault` and `DefaultSeverity`, but have tags to "overwrite" that? (e.g, for rules that we want to disable even in `All` mode)

I'm not really sure if we can start with the simpler approach of hard-coding that inside AnalyzerSeverityDecider, and implement tags later if we found a problem with hard-coding. That kinda overlaps with your previous comment as well. Let's discuss on Tuesday.